File: dotnet/RelevantCodes.ExtentReports/RelevantCodes.ExtentReports/Model/SystemProperties.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace RelevantCodes.ExtentReports.Model
{
    public class SystemProperties
    {
        public void setSystemInfo(Dictionary<String, String> info)
        {
            //if (this.info.Count() > 0)
            //info.Concat(this.info);

            //this.info = info;
            this.info.Concat(info);
        }

        public void setSystemInfo(String k, String v)
        {
            info.Add(k, v);
        }

        public Dictionary<String, String> getSystemInfo()
        {
            return info;
        }

        public SystemProperties()
        {
            info = new Dictionary<String, String>();
        }

        private Dictionary<String, String> info;
    }
}

File: src/main/java/com/relevantcodes/extentreports/model/SystemProperties.java
/*
* Copyright (c) 2015, Anshoo Arora (Relevant Codes).  All rights reserved.
* 
* Copyrights licensed under the New BSD License.
* 
* See the accompanying LICENSE file for terms.
*/

package com.relevantcodes.extentreports.model;

import java.io.Serializable;
import java.util.LinkedHashMap;
import java.util.Map;

public class SystemProperties implements Serializable {

    private static final long serialVersionUID = 8115692530725523101L;

    public void setSystemInfo(Map<String, String> info) {
        //if (this.info.size() > 0)
            //info.putAll(this.info);
        
        //this.info = info;
        this.info.putAll(info);
    }
    
    public void setSystemInfo(String k, String v) {
        info.put(k, v);
    }
    
    public Map<String, String> getSystemInfo() {
        return info;
    }
    
    public SystemProperties() {
        info = new LinkedHashMap<String, String>();        
    }
    
    private Map<String, String> info;
}
